src/controllers/searchController.js

```diff
@@ -0,0 +1,49 @@
+
+import model from "../database/models";
+import Sequelize from 'sequelize'
+
+const Op = Sequelize.Op;
+
+const returnQueries = (query) =>{
+    const queries = [];
+    for (const key in query) {
+      const value = query[key];
+      
+      if (key === 'requestStatus') queries.push({ [key]: { [Op.iLike]: `%${value}%` } }) ;
+      else queries.push( { [key]: value })
+      
+    }
+    return queries;
+}
+
+export const searchRequest = async (req, res) => {
+    const queries = returnQueries(req.query);
+    const { id } = req.userData;
+    const Request = await model.request.findAll({
```

Comment:
Try checking if the request to the database has gone successfully so you can know which error to throw if it didn't go through cz when you pass incorrect query like `?idUser=kl `the request keep going endlessly it should show bad request or another error message